### PR TITLE
[JENKINS-38711] Add UncaughtExceptionHandler to remoting related threads

### PR DESCRIPTION
**Pull request by ghost**
See [JENKINS-38711](https://issues.jenkins-ci.org/browse/JENKINS-38711).

I couldn't reopen the old PR #2986 

### Proposed changelog entries

* Entry 1: Handle uncaught exceptions in `TcpSlaveAgentListener` threads to restart them when possible.

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev @reviewbybees

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
It will still start with wrong port if the configuration changes before `TcpSlaveAgentListenerRescheduler#doAperiodicRun()`. My proposal would be to stop passing the configuration to the rescheduler job, and to just check the current setting there (and exit if the port is -1)

**Comment by rysteboe** [comment]
@oleg-nenashev sorry for long delay here, PTAL

File: core/src/main/java/hudson/TcpSlaveAgentListener.java [hunk 1]
```diff
@@ -301,6 +325,27 @@ private void error(PrintWriter out, String msg) throws IOException {
...
+    private abstract class ConnectionHandlerFailureCallback {
+        private Thread parentThread;
+        private int port;
+
+        public ConnectionHandlerFailureCallback(Thread parentThread, int port) {
+            this.parentThread = parentThread;
+            this.port = port;
+        }
+
+        public int getPort() {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/TcpSlaveAgentListener.java]: I see no usages of this `port` entry. Maybe makes sense to just remove that

Comment (rysteboe) [reply core/src/main/java/hudson/TcpSlaveAgentListener.java]: Ah yeah, remnant from before.  Fixed.

**Comment by oleg-nenashev** [comment]
Remoting update is here: https://github.com/jenkinsci/jenkins/pull/3061

**Review by oleg-nenashev — APPROVED** [review]
🐝 and @reviewbybees done
Extra reviews would be really appreciated. There are also lots of this enhancement bits in https://github.com/jenkinsci/jenkins/pull/3061

**Comment by daniel-beck** [comment]
Let's merge this towards 2.85.

**Comment by oleg-nenashev** [comment]
Agreed. Applies to every other ready PR as well

**Pull request merged**